x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/stop_crawl_popover_context_menu.test.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { EuiButton, EuiPopover } from '@elastic/eui';
+
+import { rerender } from '../../../../../test_helpers';
+
+import { StopCrawlPopoverContextMenu } from './stop_crawl_popover_context_menu';
+
+const stopCrawl = jest.fn();
+
+describe('StopCrawlsPopoverContextMenu', () => {
+  it('is initially closed', () => {
+    const wrapper = shallow(<StopCrawlPopoverContextMenu stopCrawl={stopCrawl} />);
+
+    expect(wrapper.is(EuiPopover)).toBe(true);
+    expect(wrapper.prop('isOpen')).toEqual(false);
+  });
+
+  it('can be opened to stop crawls', () => {
+    const wrapper = shallow(<StopCrawlPopoverContextMenu stopCrawl={stopCrawl} />);
+
+    wrapper.dive().find(EuiButton).simulate('click');
+    rerender(wrapper);
+
+    expect(wrapper.prop('isOpen')).toEqual(true);
+
+    // TODO I can't figure out how to find the EuiContextMenuItem inside this component's
+    // EuiContextMenuPanel.  It renders inside of an EuiResizeObserver and I figure out how
+    // to get in it
+
+    // const menuItem = wrapper
+    //   .find(EuiContextMenuPanel)
+    //   .find(EuiResizeObserver)
+    //   .find(EuiContextMenuItem);
+
+    // expect(menuItem).toHaveLength(1);
+
+    // menuItem.simulate('click');
+
+    // expect(stopCrawl).toHaveBeenCalled();
+
+    // rerender(wrapper);
+
+    // expect(wrapper.dive().find(EuiContextMenuPanel)).toHaveLength(0);
+  });
```

Comment:
This seems to work:

```ts
it('can be opened to stop crawls', () => {
  const wrapper = mount(<StopCrawlPopoverContextMenu stopCrawl={stopCrawl} />);

  wrapper.find(EuiButton).simulate('click');

  expect(wrapper.find(EuiPopover).prop('isOpen')).toEqual(true);

  const menuItem = wrapper
    .find(EuiContextMenuPanel)
    .find(EuiResizeObserver)
    .find(EuiContextMenuItem);

  expect(menuItem).toHaveLength(1);

  menuItem.simulate('click');

  expect(stopCrawl).toHaveBeenCalled();
});
```

But not sure if we are OK with not using `shallow` in that test.

<details>
<summary>Diff</summary>

```diff
diff --git a/x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/stop_crawl_popover_context_menu.test.tsx b/x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/stop_crawl_popover_context_menu.test.tsx
index 654aa9226cd..a2a30f03047 100644
--- a/x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/stop_crawl_popover_context_menu.test.tsx
+++ b/x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawler_status_indicator/stop_crawl_popover_context_menu.test.tsx
@@ -6,11 +6,15 @@
  */
 import React from 'react';
 
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 
-import { EuiButton, EuiPopover } from '@elastic/eui';
-
-import { rerender } from '../../../../../test_helpers';
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiResizeObserver,
+} from '@elastic/eui';
 
 import { StopCrawlPopoverContextMenu } from './stop_crawl_popover_context_menu';
 
@@ -25,30 +29,21 @@ describe('StopCrawlsPopoverContextMenu', () => {
   });
 
   it('can be opened to stop crawls', () => {
-    const wrapper = shallow(<StopCrawlPopoverContextMenu stopCrawl={stopCrawl} />);
-
-    wrapper.dive().find(EuiButton).simulate('click');
-    rerender(wrapper);
-
-    expect(wrapper.prop('isOpen')).toEqual(true);
-
-    // TODO I can't figure out how to find the EuiContextMenuItem inside this component's
-    // EuiContextMenuPanel.  It renders inside of an EuiResizeObserver and I figure out how
-    // to get in it
+    const wrapper = mount(<StopCrawlPopoverContextMenu stopCrawl={stopCrawl} />);
 
-    // const menuItem = wrapper
-    //   .find(EuiContextMenuPanel)
-    //   .find(EuiResizeObserver)
-    //   .find(EuiContextMenuItem);
+    wrapper.find(EuiButton).simulate('click');
 
-    // expect(menuItem).toHaveLength(1);
+    expect(wrapper.find(EuiPopover).prop('isOpen')).toEqual(true);
 
-    // menuItem.simulate('click');
+    const menuItem = wrapper
+      .find(EuiContextMenuPanel)
+      .find(EuiResizeObserver)
+      .find(EuiContextMenuItem);
 
-    // expect(stopCrawl).toHaveBeenCalled();
+    expect(menuItem).toHaveLength(1);
 
-    // rerender(wrapper);
+    menuItem.simulate('click');
 
-    // expect(wrapper.dive().find(EuiContextMenuPanel)).toHaveLength(0);
+    expect(stopCrawl).toHaveBeenCalled();
   });
 });
```
</details>

Reply:
💯 thanks @orhantoy